packages/contentful--create-contentful-app/src/template.ts

```diff
@@ -4,29 +4,102 @@ import degit from 'degit';
 import rimraf from 'rimraf';
 
 import { CLIOptions, ContentfulExample } from './types';
-import { highlight, warn } from './logger';
+import { highlight, success, warn, wrapInBlanks } from './logger';
 import { rmIfExists } from './utils';
+// @ts-expect-error no types available
+import fetch from 'node-fetch';
+import inquirer from 'inquirer';
+import chalk from 'chalk';
+
+async function getGithubFolderNames(username: string, repo: string, path: string) {
+  const url = `https://api.github.com/repos/${username}/${repo}/contents/${path}`;
+
+  const response = await fetch(url);
+  const contents: any = await response.json();
+
+  return contents.filter((content: any) => content.type === "dir")
+  .map((content: any) => content.name);
+}
+  
+  
+async function promptExampleSelection(): Promise<string> {
+  let template = 'typescript'
+
+  // ask user whether to start with an empty template or use an example template
+  const { starter } = await inquirer.prompt([
+    {
+      name: 'starter',
+      message: 'Do you want to start from an example template or use the empty template?',
+      type: 'list',
+      choices: ['empty', 'template'],
+      default: 'empty',
+    },
+  ]);
+
+  // if the user chose to use an empty template, ask which language they prefer
+  if (starter === "empty") {
+    const { language } = await inquirer.prompt([
+      {
+        name: 'language',
+        message: 'Do you prefer Typescript or Javascript',
+        type: 'list',
+        choices: ['javascript', 'typescript'],
+        default: 'typescript',
+      },
+    ]);
+    template = language
+  } else {
+    // get available templates from examples
+    const availableTemplates = await getGithubFolderNames("contentful", "apps", "examples")
+    
+    // ask user to select a template from the available examples
+    const { example } = await inquirer.prompt([
+      {
+        name: 'example',
+        message: 'Select a template',
+        type: 'list',
+        choices: availableTemplates,
+      },
+    ]);
+
+    template = example
+  }
+
+  // return the selected template
+  return selectTemplate(template)
+}
 
 const EXAMPLES_PATH = 'contentful/apps/examples/';
```

Comment:
nit: would be nice to have var's in the beginning of the file

Reply:
👍